[FEAT/#48] 온보딩 만들었어요

angryPodo opened the pull request:
## Related issue 🛠
- closed #48

## Work Description ✏️
- 온보딩 스크린 구현
- 네비게이션 만듬
- 텍스트 필드 로직 살짝 수정

## Screenshot 📸

https://github.com/user-attachments/assets/5bdea1b1-580a-4516-8ba8-95015199c4dd


## To Reviewers 📢
뷰모델은 서버 연결과 함께 같이 만들겠습니다 ㅎㅎ

angryPodo committed:
[feat/#48] 온보딩 모듈 생성

angryPodo committed:
[feat/#48] 온보딩 화면 구현 및 네비게이션 세팅

Hyobeen-Park reviewed (commented):
spacer에 weight값은 기준이 뭐에요?????????????????

Hyobeen-Park reviewed (approved):
답변은 오프라인으로 들었으니까 어푸 드릴게요ㅋㅋㅋㅋ 어푸푸🚀🚀

angryPodo committed:
Merge remote-tracking branch 'origin/develop' into feat/#48-onboarding-screen

# Conflicts:
#	presentation/main/src/main/java/com/hilingual/presentation/main/MainScreen.kt

angryPodo committed:
[MOD/#48] Modify navigation flow from AuthScreen to OnboardingScreen

The navigation flow has been updated. Previously, a successful Google sign-in on `AuthScreen` would navigate directly to `Home`. This has been changed so that it now navigates to `OnboardingScreen` first.

Key changes:
- `AuthScreen`: `onGoogleSignClick` now calls `navigateToOnboarding` instead of `navigateToHome`.
- `OnboardingScreen`:
    - The status bar color is now set to white.
    - The default profile image now has a circular border.
- `MainScreen`: The `navigateToOnboarding` function is now passed to the `authNavGraph`.
- `MainNavigator`: Added a `navigateToOnboarding` function to handle navigation to the onboarding flow.

nhyeonii reviewed (approved):
오프라인 코드 확인 완룧ㅎㅎ 수고하셧습니다 ~~

Daljyeong reviewed (approved):
코드 확인 완료~~ 수고하셨숩니당!! 😎

nahy-512 commented on presentation/onboarding/src/main/java/com/hilingual/presentation/onboarding/OnboardingScreen.kt:
```diff
@@ -0,0 +1,154 @@
+    paddingValues: PaddingValues,
+    value: String,
+    onValueChanged: (String) -> Unit,
+    isValid: () -> Boolean,
```
p3
이 부분이 [이거](https://angrypodo.tistory.com/10) 적용하신 거 맞죠?

위에 value는 왜
```
value: () -> String,
```
으로 안하셨나여

angryPodo replied:
아뇨 지영씨가 만든거 그냥 가져다가 썼는데요ㅋㅋ 지금은 뷰 조립 PR이라서 로직이고 뭐고 UI만 만들었습니다ㅎㅎ

nahy-512 replied:
아ㅋㅋㅋ 확인입니다
어쩐지 이슈 팔 때 '뷰'라고 강조를 해주셨군여

angryPodo commented:
여러분 근데 `graphemeLength`이거 이해하고 어푸 누른거 맞죠? @Hi-lingual/team-android

Daljyeong commented:
네엥 근데 궁금한 게 종종 이모티콘을 1글자가 아닌 2글자로 처리하는 경우도 있던데(ex. 카카오톡 선물하기에서 편지 작성할 때의 이모티콘 글자 수 처리) 이모티콘 글자 수 처리에 대한 내용이 기능 명세서에 있었나용??

angryPodo commented:
그건 카카오가 좀 특이하네요..보통 1글자로 세기 위해서 그래핌 단위로 쪼개는건데...왜 2글자로 치는거지????

nahy-512 commented on core/designsystem/src/main/java/com/hilingual/core/designsystem/component/textfield/HilingualShortTextField.kt:
```diff
@@ -50,7 +50,7 @@ fun HilingualShortTextField(
             textStyle = HilingualTheme.typography.bodyM16,
             onValueChanged = {
-                if (it.length <= maxLength) onValueChanged(it)
+                if (it.graphemeLength <= maxLength) onValueChanged(it)
```
p3
이 `graphemeLength`는 한글 글자수를 제대로 세기 위한 용도일까요??

nahy-512 replied:
어라.. 확인하는 동안 위에 코멘트 다셨네요. 아무래도 전 이해 못한 듯🥲

nahy-512 replied:
이모지 사용 한글자로 체크하기??

angryPodo replied:
이모지가 있는 문자열 .length 로 찍어보면 글자수가 이상할겁니다ㅎㅎ 이걸 해결하기 위한 방법이 여러가지 있으니 공부해보십시오

nahy-512 replied:
엉엉.. 너무 어려워요 미미나 열어주세요 !!

nahy-512 reviewed (approved):
LGTM~~🚀
아직 닉네임 에러 처리 안 해주신 걸로 보이는데, 이런 부분은 `Uncompleted Tasks`로 설명해주면 코드 확인할 때 조금 더 편할 거 같아요ㅎ.ㅎ
이번 작업도 고생하셨습니다~!

Hyobeen-Park commented:
> 그건 카카오가 좀 특이하네요..보통 1글자로 세기 위해서 그래핌 단위로 쪼개는건데...왜 2글자로 치는거지????

사실 카카오도 이모지 처리는 귀찮았던거 아님?ㄷㄷ

angryPodo commented:
> 아직 닉네임 에러 처리 안 해주신 걸로 보이는데, 이런 부분은 Uncompleted Tasks로 설명해주면 코드 확인할 때 조금 더 편할 거 같아요ㅎ.ㅎ
이번 작업도 고생하셨습니다~!

이건 다음 PR에서~ 요 PR은 UI와 짜바리 PR입니다

Daljyeong commented:
> > 그건 카카오가 좀 특이하네요..보통 1글자로 세기 위해서 그래핌 단위로 쪼개는건데...왜 2글자로 치는거지????
> 
> 사실 카카오도 이모지 처리는 귀찮았던거 아님?ㄷㄷ

옛날 메시지 다시 보면서 정신 차리고 계산해보니 카톡에서도 특수문자 1글자로 처리되고 있었네요...하핳... 기억의 오류였을 뿐...
헷갈리게 해서 미안함니다..ㅎㅎㅎ 😨😰😱🫠
